src/Files.App/Controllers/TerminalController.cs

```diff
@@ -15,198 +15,192 @@
+			.AsTask());
+
+			if (Folder is null)
+			{
+				model = await GetDefaultTerminalFileModel();
+				return;
+			}
+
+			var jsonFile = await FilesystemTasks.Wrap(() =>
+				Folder.GetFileAsync(JsonFileName)
+			.AsTask());
+
+			if (!jsonFile)
+			{
+				model = await GetDefaultTerminalFileModel();
+				if (jsonFile.ErrorCode is FileSystemStatusCode.NotFound)
+					SaveModel();
+				return;
+			}
+
+			try
+			{
+				configContent = await FileIO.ReadTextAsync(jsonFile.Result);
+
+				var fileModel = JsonSerializer.Deserialize<TerminalFileModel>(configContent);
+				if (fileModel is null)
+					throw new ArgumentException($"{JsonFileName} is empty, regenerating...");
+				model = fileModel!;
+			}
+			catch (ArgumentException)
+			{
+				model = await GetDefaultTerminalFileModel();
+				SaveModel();
+			}
+			catch (Exception)
+			{
+				model = await GetDefaultTerminalFileModel();
+			}
+		}
+
+		private async Task StartWatchConfigChangeAsync()
+		{
+			var queryOptions = new QueryOptions
+			{
+				ApplicationSearchFilter = "System.FileName:" + JsonFileName
+			};
+
+			var settingsFolder = await ApplicationData.Current.LocalFolder.GetFolderAsync("settings");
+			var query = settingsFolder.CreateFileQueryWithOptions(queryOptions);
+			query.ContentsChanged += Query_ContentsChanged;
+			await query.GetFilesAsync();
+		}
+
+		private async void Query_ContentsChanged(IStorageQueryResultBase sender, object args)
+		{
+			try
+			{
+				var uri = new Uri("ms-appdata:///local/settings/terminal.json");
+				var file = await StorageFile.GetFileFromApplicationUriAsync(uri);
+				var content = await FileIO.ReadTextAsync(file);
+
+				if (configContent == content)
+					return;
+
+				configContent = content;
+				await LoadAsync();
+				await GetInstalledTerminalsAsync(model);
+				SaveModel();
+				ModelChanged?.Invoke(this);
+			}
+			catch {}
+		}
+
+		private async Task<TerminalFileModel> GetDefaultTerminalFileModel()
+		{
+			try
+			{
+				var uri = new Uri(defaultTerminalPath);
+				var defaultFile = await StorageFile.GetFileFromApplicationUriAsync(uri);
+				var defaultContent = await FileIO.ReadTextAsync(defaultFile);
+
+				var fileModel = JsonSerializer.Deserialize<TerminalFileModel>(configContent);
+				if (fileModel is null)
+					throw new ArgumentException($"{JsonFileName} is empty, regenerating...");
+				model = fileModel!;
+
+				await GetInstalledTerminalsAsync(model);
+				model.ResetToDefaultTerminal();
+				return model;
+			}
+			catch
+			{
+				var model = new TerminalFileModel();
+				await GetInstalledTerminalsAsync(model);
+				model.ResetToDefaultTerminal();
+				return model;
+			}
+		}
+
+		private static async Task GetInstalledTerminalsAsync(TerminalFileModel model)
+		{
+			var terminalDefs = new Dictionary<Terminal, bool>
+			{
+				{
+					new Terminal
+					{
+						Name = "Windows Terminal",
+						Path = "wt.exe",
+						Arguments = "-d .",
+						Icon = string.Empty,
+					},
```

Comment:
Maybe overkill, but I'd suggest to create a method that construct the Terminal instance with given arguments. In case we want to add someday an argument to the process, we will have only to modify one function rather than three constructors.

Reply:
I modified the Terminal class to add constructors and fixed the warnings.